pyteal/ast/router.py

```diff
@@ -339,52 +339,83 @@ def wrap_handler(
                     f"got {handler.subroutine.argument_count()} args with {len(handler.subroutine.abi_args)} ABI args."
                 )
 
+            # All subroutine args types
             arg_type_specs = cast(
                 list[abi.TypeSpec], handler.subroutine.expected_arg_types
             )
-            if handler.subroutine.argument_count() > METHOD_ARG_NUM_CUTOFF:
-                last_arg_specs_grouped = arg_type_specs[METHOD_ARG_NUM_CUTOFF - 1 :]
-                arg_type_specs = arg_type_specs[: METHOD_ARG_NUM_CUTOFF - 1]
-                last_arg_spec = abi.TupleTypeSpec(*last_arg_specs_grouped)
-                arg_type_specs.append(last_arg_spec)
-
-            arg_abi_vars: list[abi.BaseType] = [
-                type_spec.new_instance() for type_spec in arg_type_specs
+
+            # All subroutine arg values, initialize here and use below instead of
+            # creating new instances on the fly so we dont have to think about splicing
+            # back in the transaction types
+            arg_vals = [typespec.new_instance() for typespec in arg_type_specs]
+
+            # Only args that appear in app args
+            app_arg_vals: list[abi.BaseType] = [
+                ats for ats in arg_vals if not isinstance(ats, abi.Transaction)
             ]
```

Comment:
this is filtered only on the assumption that, there are no transaction types existing in underlying composite type spec, right?